### ISSUE TITLE
Android :     Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.quickblox.chat.QBVideoChatWebRTCSignalingManager.addSignalingManagerListener

### ISSUE THREAD

**Issue by iosgnanavel** [QuickBlox/quickblox-android-sdk#441]
Hi Team,
 **Following Exception occur while start CallService in QB.. How can i fix this issue.** 
**Issue :**Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.quickblox.chat.QBVideoChatWebRTCSignalingManager.addSignalingManagerListener(com.quickblox.chat.listeners.QBVideoChatSignalingManagerListener)' on a null object reference
       at com.wtf.services.CallService.initQBRTCClient(CallService.java:149)
       at com.wtf.services.CallService.startActionsOnSuccessLogin(CallService.java:129)
       at com.wtf.services.CallService.access$100(CallService.java:34)
       at com.wtf.services.CallService$1.onSuccess(CallService.java:114)
       at com.wtf.services.CallService$1.onSuccess(CallService.java:110)
       at com.quickblox.core.helper.CommonUtils.notifyEntityCallbackOnSuccess(Unknown Source)
       at com.quickblox.chat.QBChatService$4.run(Unknown Source)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Source Code:**   

  private void initQBRTCClient() {
        rtcClient = QBRTCClient.getInstance(getApplicationContext());
        // Add signalling manager
        if (chatService == null) {
            createChatService();
        }
        _

> chatService.getVideoChatWebRTCSignalingManager().addSignalingManagerListener(new QBVideoChatSignalingManagerListener() {

_
            @Override
            public void signalingCreated(QBSignaling qbSignaling, boolean createdLocally) {
                if (!createdLocally) {
                    rtcClient.addSignaling((QBWebRTCSignaling) qbSignaling);
                }
            }
        });

        // Configure
        QBRTCConfig.setDebugEnabled(true);
        SettingsUtil.configRTCTimers(CallService.this);

        // Add service as callback to RTCClient
        rtcClient.addSessionCallbacksListener(WebRtcSessionManager.getInstance(this));
        rtcClient.prepareToProcessCalls();
    }

**Comment by RomanPronin** [comment]
Hi @iosgnanavel, you should be login in chat before getVideoChatWebRTCSignalingManager, 
`if (!chatService.isLoggedIn()) {
           chatService.login(QBUser user);}`

**Comment by petergayduchik** [comment]
this error still appears. Its randomly make crash. i also check isLoggedIn

**Comment by RomanPronin** [comment]
Hi, @petergayduchik, what version SDK do you use? Is this issue reproducible on [videosample](https://github.com/QuickBlox/quickblox-android-sdk/tree/master/sample-videochat-webrtc)?

**Comment by petergayduchik** [comment]
@RomanPronin 3.3.4 im using.
```
    compile 'com.quickblox:quickblox-android-sdk-chat:3.3.4'
    compile 'com.quickblox:quickblox-android-sdk-core:3.3.4'
    compile 'com.quickblox:quickblox-android-sdk-messages:3.3.4'
    compile 'com.quickblox:quickblox-android-sdk-videochat-webrtc:3.3.4'
    compile project(':quickblox-android-sdk-content-3.3.4')
```
this error realy randomly showing. I cant reproduce this bug. When i forgot about this error - error come to my head -_-
`java.lang.NullPointerException: Attempt to invoke virtual method 'void com.quickblox.chat.QBVideoChatWebRTCSignalingManager.addSignalingManagerListener(com.quickblox.chat.listeners.QBVideoChatSignalingManagerListener)' on a null object reference`

on line 
```
QBChatService.getInstance().getVideoChatWebRTCSignalingManager()
                    .addSignalingManagerListener(new QBVideoChatSignalingManagerListener() {
                        @Override
                        public void signalingCreated(QBSignaling qbSignaling, boolean createdLocally) {
                            if (!createdLocally) {
                                QBRTCClient.getInstance(context).addSignaling((QBWebRTCSignaling) qbSignaling);
                                Logger.d("add signalling ");
                            }
                        }
                    });
```

here
` .addSignalingManagerListener(new QBVideoChatSignalingManagerListener() {`

before i see this error in logcat - its take to time. Like app loading... bla bla errors also here

`org.jivesoftware.smack.SmackException$ConnectionException: The following addresses failed: 'chat.quickblox.com:5223' failed because java.net.SocketTimeoutException: failed to connect to chat.quickblox.com/34.192.242.204 (port 5223) after 30000ms`

sure its can be internet access problem. I just cant reproduce. Everything what i have - i show here.

**Comment by RomanPronin** [comment]
If you successfully logged in to chat, before addSignalingManagerListener(), and still have a issue, in that case, I can offer you to check qbVideoChatWebRTCSignalingManager != null, before adding listener, to avoid NullPointerException.

**Comment by islavstan** [comment]
### Hello! I have the same issue. I use your sample-videochat-webrtc. In CallService 
 private void loginToChat(QBUser qbUser) {
        chatService.login(qbUser, new QBEntityCallback<QBUser>() {
            @Override
            public void onSuccess(QBUser qbUser, Bundle bundle) {
                Log.d(TAG, "login onSuccess");
                startActionsOnSuccessLogin();
            }

            @Override
            public void onError(QBResponseException e) {
                Log.d(TAG, "login onError " + e.getMessage());
                sendResultToActivity(false, e.getMessage() != null
                        ? e.getMessage()
                        : "Login error");
            }
        });
    }
### method startActionsOnSuccessLogin call only after success login. In method  initQBRTCClient()
  private void initQBRTCClient() {
        rtcClient = QBRTCClient.getInstance(getApplicationContext());
        // Add signalling manager
        if (chatService != null && chatService.getVideoChatWebRTCSignalingManager() != null) {
            chatService.getVideoChatWebRTCSignalingManager().addSignalingManagerListener((qbSignaling, createdLocally) -> {
                if (!createdLocally) {
                    rtcClient.addSignaling((QBWebRTCSignaling) qbSignaling);
                }
            });
        }
        // Configure
        QBRTCConfig.setDebugEnabled(true);
        SettingsUtil.configRTCTimers(CallService.this);

        // Add service as callback to RTCClient
        rtcClient.addSessionCallbacksListener(WebRtcSessionManager.getInstance(this));
        rtcClient.prepareToProcessCalls();
    }
### I add two check for null, but this did not solve the problem

Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'com.quickblox.chat.QBWebRTCSignaling com.quickblox.chat.QBVideoChatWebRTCSignalingManager.getSignaling(int)' on a null object reference
       at com.quickblox.videochat.webrtc.QBRTCSession.initSignallingWithOpponents(QBRTCSession.java:950)
       at com.quickblox.videochat.webrtc.QBRTCSession.<init>(QBRTCSession.java:141)
       at com.quickblox.videochat.webrtc.QBRTCClient.createNewSessionWithOpponents(QBRTCClient.java:168)
       at biz.vomer.voip.base.MainActivity$5.onPermissionGranted(MainActivity.java:706)
       at com.master.permissionhelper.PermissionHelper.request(PermissionHelper.java:80)
       at biz.vomer.voip.base.MainActivity.startCall(MainActivity.java:697)
       at biz.vomer.voip.screens.chat.ChatFragment.lambda$onViewCreated$6(ChatFragment.java:341)
       at biz.vomer.voip.screens.chat.ChatFragment$$Lambda$7.onClick(Unknown Source)
       at android.view.View.performClick(View.java:4848)
       at android.view.View$PerformClick.run(View.java:20262)
       at android.os.Handler.handleCallback(Handler.java:815)
       at android.os.Handler.dispatchMessage(Handler.java:104)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5651)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
### In your library I have problems only with WebRTCSignalingManager, please check

**Comment by tatanka987** [comment]
@islavstan  please provide full log from start login to the chat to getting this error for better understanding your issue (chat's logs must be enabled QBChatService.setDebugEnabled(true);)

**Comment by megahacker15** [comment]
I have the same issue. I provide my log 

11-19 21:33:44.625 20464-20464/? I/art: Late-enabling -Xcheck:jni
11-19 21:33:44.818 20464-20464/call.biz.vomer.voip D/Proxy: setHttpRequestCheckHandler
11-19 21:33:45.928 20464-20464/call.biz.vomer.voip I/MultiDex: VM with version 2.1.0 has multidex support
11-19 21:33:45.928 20464-20464/call.biz.vomer.voip I/MultiDex: install
11-19 21:33:45.928 20464-20464/call.biz.vomer.voip I/MultiDex: VM has multidex support, MultiDex support library is disabled.
11-19 21:33:45.972 20464-20464/call.biz.vomer.voip D/FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
11-19 21:33:46.020 20464-20464/call.biz.vomer.voip D/FirebaseApp: com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
11-19 21:33:46.110 20464-20464/call.biz.vomer.voip I/FA: App measurement is starting up, version: 11400
11-19 21:33:46.111 20464-20464/call.biz.vomer.voip I/FA: To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
11-19 21:33:46.169 20464-20464/call.biz.vomer.voip V/FA: Collection enabled
11-19 21:33:46.170 20464-20464/call.biz.vomer.voip V/FA: App package, google app id: call.biz.vomer.voip, 1:593952674245:android:416e593acf288561
11-19 21:33:46.171 20464-20464/call.biz.vomer.voip I/FA: To enable faster debug mode event logging run:
                                                           adb shell setprop debug.firebase.analytics.app call.biz.vomer.voip
11-19 21:33:46.171 20464-20464/call.biz.vomer.voip D/FA: Debug-level message logging enabled
11-19 21:33:46.246 20464-20464/call.biz.vomer.voip V/FA: Registered activity lifecycle callback
11-19 21:33:46.248 20464-20464/call.biz.vomer.voip I/FirebaseInitProvider: FirebaseApp initialization successful
11-19 21:33:46.251 20464-20464/call.biz.vomer.voip I/InstantRun: starting instant run server: is main process
11-19 21:33:46.253 20464-20557/call.biz.vomer.voip V/FA: Persisting first open: 1511120026079
11-19 21:33:46.380 20464-20464/call.biz.vomer.voip I/CrashlyticsCore: Initializing Crashlytics 2.4.1.19
11-19 21:33:46.403 20464-20561/call.biz.vomer.voip W/ResourcesManager: Asset path '/system/framework/com.android.media.remotedisplay.jar' does not exist or contains no resources.
11-19 21:33:46.403 20464-20561/call.biz.vomer.voip W/ResourcesManager: Asset path '/system/framework/com.android.location.provider.jar' does not exist or contains no resources.
11-19 21:33:46.456 20464-20464/call.biz.vomer.voip D/QBASDK: Restored custom endpoints. ApiEndpoint: , ChatEndpoint: . lastUpdateTime: Thu Jan 01 03:00:00 GMT+02:00 1970
11-19 21:33:46.459 20464-20464/call.biz.vomer.voip D/QBASDK: DEVICE_ID = ffffffff-9635-fa42-7d20-a8af288ce5e5
11-19 21:33:46.509 20464-20464/call.biz.vomer.voip D/QBASDK: creating app secured preferences
11-19 21:33:46.919 20464-20557/call.biz.vomer.voip V/FA: State of service unknown
11-19 21:33:46.919 20464-20557/call.biz.vomer.voip V/FA: Checking service availability
11-19 21:33:46.921 20464-20557/call.biz.vomer.voip V/FA: Service available
11-19 21:33:46.921 20464-20557/call.biz.vomer.voip V/FA: Setting useService: true
11-19 21:33:46.922 20464-20557/call.biz.vomer.voip V/FA: Using measurement service
11-19 21:33:46.937 20464-20557/call.biz.vomer.voip V/FA: Connecting to remote service
11-19 21:33:46.952 20464-20557/call.biz.vomer.voip V/FA: Using measurement service
11-19 21:33:46.952 20464-20557/call.biz.vomer.voip V/FA: Connection attempt already in progress
11-19 21:33:48.194 20464-20464/call.biz.vomer.voip I/MultiDex: install
11-19 21:33:48.194 20464-20464/call.biz.vomer.voip I/MultiDex: VM has multidex support, MultiDex support library is disabled.
11-19 21:33:48.196 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.196 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.196 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.197 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.197 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.197 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.197 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.198 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.198 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.198 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.199 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.199 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<org.greenrobot.greendao.database.DatabaseOpenHelper$EncryptedHelper>
11-19 21:33:48.238 20464-20464/call.biz.vomer.voip I/greenDAO: Creating tables for schema version 1
11-19 21:33:48.330 20464-20576/call.biz.vomer.voip I/stetho: Listening on @stetho_call.biz.vomer.voip_devtools_remote
11-19 21:33:48.528 20464-20464/call.biz.vomer.voip W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
11-19 21:33:48.606 20464-20464/call.biz.vomer.voip V/FA: onActivityCreated
11-19 21:33:48.607 20464-20464/call.biz.vomer.voip I/Timeline: Timeline: Activity_launch_request id:call.biz.vomer.voip time:167387909
11-19 21:33:48.676 20464-20561/call.biz.vomer.voip W/ResourcesManager: Asset path '/system/framework/com.android.media.remotedisplay.jar' does not exist or contains no resources.
11-19 21:33:48.676 20464-20561/call.biz.vomer.voip W/ResourcesManager: Asset path '/system/framework/com.android.location.provider.jar' does not exist or contains no resources.
11-19 21:33:48.679 20464-20487/call.biz.vomer.voip I/art: Background sticky concurrent mark sweep GC freed 43878(1696KB) AllocSpace objects, 1(20KB) LOS objects, 0% free, 17MB/17MB, paused 3.872ms total 116.117ms
11-19 21:33:48.753 20464-20464/call.biz.vomer.voip V/FA: onActivityCreated
11-19 21:33:48.776 20464-20487/call.biz.vomer.voip I/art: Background partial concurrent mark sweep GC freed 34437(1120KB) AllocSpace objects, 1(3MB) LOS objects, 39% free, 12MB/20MB, paused 6.417ms total 92.276ms
11-19 21:33:48.819 20464-20464/call.biz.vomer.voip D/ContextImpl: isCts() value : -1
11-19 21:33:48.819 20464-20464/call.biz.vomer.voip D/ContextImpl: isCts()  package name : call.biz.vomer.voip
11-19 21:33:48.819 20464-20464/call.biz.vomer.voip D/wangcy9: setStatusIcon occur wrong theme!
11-19 21:33:49.179 20464-20464/call.biz.vomer.voip I/TextInputLayout: EditText added is not a TextInputEditText. Please switch to using that class instead.
11-19 21:33:49.271 20464-20464/call.biz.vomer.voip I/PermissionHelper: PERMISSION: Permission Granted
11-19 21:33:49.362 20464-20557/call.biz.vomer.voip V/FA: Using measurement service
11-19 21:33:49.362 20464-20557/call.biz.vomer.voip V/FA: Connection attempt already in progress
11-19 21:33:49.366 20464-20557/call.biz.vomer.voip V/FA: Activity resumed, time: 510241990
11-19 21:33:49.369 20464-20590/call.biz.vomer.voip D/OpenGLRenderer: Render dirty regions requested: true
11-19 21:33:49.370 20464-20557/call.biz.vomer.voip I/FA: Tag Manager is not found and thus will not be used
11-19 21:33:49.381 20464-20464/call.biz.vomer.voip D/Atlas: Validating map...
11-19 21:33:49.384 20464-20557/call.biz.vomer.voip D/FA: Logging event (FE): screen_view(_vs), Bundle[{firebase_event_origin(_o)=auto, firebase_screen_class(_sc)=MainActivity, firebase_screen_id(_si)=1941906671156581182}]
11-19 21:33:49.443 20464-20557/call.biz.vomer.voip V/FA: Using measurement service
11-19 21:33:49.443 20464-20557/call.biz.vomer.voip V/FA: Connection attempt already in progress
11-19 21:33:49.444 20464-20557/call.biz.vomer.voip D/FA: Connected to remote service
11-19 21:33:49.444 20464-20557/call.biz.vomer.voip V/FA: Processing queued up service tasks: 4
11-19 21:33:49.471 20464-20590/call.biz.vomer.voip I/Adreno-EGL: <qeglDrvAPI_eglInitialize:379>: EGL 1.4 QUALCOMM build: AU_LINUX_ANDROID_LA.BR.1.1.2_RB1.05.00.02.031.023_msm8916_64_LA.BR.1.1.2_RB1__release_AU (Iba0e932ba2)
                                                                 OpenGL ES Shader Compiler Version: E031.25.03.02
                                                                 Build Date: 03/23/15 Mon
                                                                 Local Branch: mybranch8505898
                                                                 Remote Branch: quic/LA.BR.1.1.2_rb1.14
                                                                 Local Patches: NONE
                                                                 Reconstruct Branch: AU_LINUX_ANDROID_LA.BR.1.1.2_RB1.05.00.02.031.023 + 491a26f + 6f948ca +  NOTHING
11-19 21:33:49.527 20464-20590/call.biz.vomer.voip I/OpenGLRenderer: Initialized EGL, version 1.4
11-19 21:33:49.552 20464-20590/call.biz.vomer.voip D/OpenGLRenderer: Enabling debug mode 0
11-19 21:33:49.860 20464-20464/call.biz.vomer.voip I/Timeline: Timeline: Activity_idle id: android.os.BinderProxy@276835c0 time:167389162
11-19 21:33:54.937 20464-20557/call.biz.vomer.voip V/FA: Inactivity, disconnecting from the service
11-19 21:33:55.599 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.support.v7.widget.AppCompatEditText{f3d1311 VFED..CL .F...... 0,0-656,91 #7f090055 app:id/et_phone}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=353.46463, y[0]=64.33331, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167394897, downTime=167394897, deviceId=7, source=0x1002 }
11-19 21:33:55.625 20464-20464/call.biz.vomer.voip I/View: Touch up dispatch to android.support.v7.widget.AppCompatEditText{f3d1311 VFED..CL .F.P.... 0,0-656,91 #7f090055 app:id/et_phone}, event = MotionEvent { action=ACTION_UP, id[0]=0, x[0]=353.46463, y[0]=64.33331, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167394926, downTime=167394897, deviceId=7, source=0x1002 }
11-19 21:33:59.367 20464-20557/call.biz.vomer.voip V/FA: Session started, time: 510251995
11-19 21:33:59.370 20464-20557/call.biz.vomer.voip D/FA: Logging event (FE): session_start(_s), Bundle[{firebase_event_origin(_o)=auto, firebase_screen_class(_sc)=MainActivity, firebase_screen_id(_si)=1941906671156581182}]
11-19 21:33:59.391 20464-20557/call.biz.vomer.voip V/FA: Using measurement service
11-19 21:33:59.391 20464-20557/call.biz.vomer.voip V/FA: Connecting to remote service
11-19 21:33:59.397 20464-20557/call.biz.vomer.voip D/FA: Connected to remote service
11-19 21:33:59.397 20464-20557/call.biz.vomer.voip V/FA: Processing queued up service tasks: 1
11-19 21:34:02.898 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:03.085 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:04.376 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:04.427 20464-20557/call.biz.vomer.voip V/FA: Inactivity, disconnecting from the service
11-19 21:34:04.688 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:05.040 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:05.517 20464-20464/call.biz.vomer.voip W/IInputConnectionWrapper: sendKeyEvent on inactive InputConnection
11-19 21:34:06.275 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.support.v7.widget.AppCompatTextView{272ca44e V.ED.... ........ 557,37-636,75}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=67.13318, y[0]=7.373169, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167405574, downTime=167405574, deviceId=7, source=0x1002 }
11-19 21:34:06.275 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.widget.RelativeLayout{2a65e96f V.E...C. ........ 0,708-720,820 #7f0900b3 app:id/rlNext}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=624.1332, y[0]=44.37317, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167405574, downTime=167405574, deviceId=7, source=0x1002 }
11-19 21:34:06.359 20464-20464/call.biz.vomer.voip I/View: Touch up dispatch to android.widget.RelativeLayout{2a65e96f V.E...C. ...P.... 0,708-720,820 #7f0900b3 app:id/rlNext}, event = MotionEvent { action=ACTION_UP, id[0]=0, x[0]=624.1332, y[0]=44.37317, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167405661, downTime=167405574, deviceId=7, source=0x1002 }
11-19 21:34:06.374 20464-20464/call.biz.vomer.voip I/PermissionHelper: PERMISSION: Permission Granted
11-19 21:34:06.515 20464-20754/call.biz.vomer.voip W/Java7Support: Unable to load JDK7 types (annotations, java.nio.file.Path): no Java7 support added
11-19 21:34:07.421 20464-20464/call.biz.vomer.voip D/wangcy9: setStatusIcon occur wrong theme!
11-19 21:34:11.344 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.support.v7.widget.AppCompatButton{963f588 VFED..C. ........ 468,8-596,104 #1020019 android:id/button1}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=101.140076, y[0]=52.553467, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167410626, downTime=167410626, deviceId=7, source=0x1002 }
11-19 21:34:11.411 20464-20464/call.biz.vomer.voip I/View: Touch up dispatch to android.support.v7.widget.AppCompatButton{963f588 VFED..C. ...p.... 468,8-596,104 #1020019 android:id/button1}, event = MotionEvent { action=ACTION_UP, id[0]=0, x[0]=101.140076, y[0]=52.553467, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167410712, downTime=167410626, deviceId=7, source=0x1002 }
11-19 21:34:11.419 20464-20464/call.biz.vomer.voip I/PermissionHelper: PERMISSION: Permission Granted
11-19 21:34:11.696 20464-20464/call.biz.vomer.voip D/Request build:: application_id=58733&auth_key=6W2dL4nn5SVcpHS&nonce=58840027&timestamp=1511120051
11-19 21:34:11.710 20464-20464/call.biz.vomer.voip D/QBASDK: =========================================================
                                                             === REQUEST ==== 540e7950-945d-42d5-a519-8fc099334627 ===
                                                             REQUEST
                                                                 POST https://api.quickblox.com/session.json
                                                             HEADERS
                                                                 QuickBlox-REST-API-Version=0.1.1
                                                                 QB-SDK=Android 3.4.1
                                                             PARAMETERS
                                                                 application_id=58733
                                                                 auth_key=6W2dL4nn5SVcpHS
                                                                 nonce=58840027
                                                                 timestamp=1511120051
                                                                 signature=75100685af55fc26ed02b5daf0575ce7ebdd8ca3
                                                             INLINE
                                                                 POST https://api.quickblox.com/session.json?application_id=58733&auth_key=6W2dL4nn5SVcpHS&nonce=58840027&timestamp=1511120051&signature=75100685af55fc26ed02b5daf0575ce7ebdd8ca3
11-19 21:34:12.768 20464-20464/call.biz.vomer.voip D/QBASDK: *********************************************************
                                                             *** RESPONSE *** 540e7950-945d-42d5-a519-8fc099334627 ***
                                                             STATUS : 201 
                                                             HEADERS
                                                                 X-Runtime=0.012444
                                                                 Cache-Control=max-age=0, private, must-revalidate
                                                                 QuickBlox-REST-API-Version=0.1.1
                                                                 Set-Cookie=_mkra_ctxt=6ef6b97b9459393fefeb2ad78b995e89--201; path=/; max-age=5; HttpOnly; secure
                                                                 QB-Token-ExpirationDate=2017-11-19 21:34:11 UTC
                                                                 Status=201 Created
                                                                 X-Request-Id=3b3f9315-3f2b-46a6-9bb8-18dea1b8e268
                                                                 X-Android-Response-Source=NETWORK 201
                                                                 Connection=Close
                                                                 X-Android-Sent-Millis=1511120052488
                                                                 Content-Type=application/json; charset=utf-8
                                                                 Date=Sun, 19 Nov 2017 19:34:11 GMT
                                                                 X-Content-Type-Options=nosniff
                                                                 Server=openresty/1.9.15.1
                                                                 ETag=W/"d8ca4e0298baabe445299bbab7c771c2"
                                                                 Content-Length=237
                                                                 Strict-Transport-Security=max-age=31536000max-age=15768000;
                                                                 X-XSS-Protection=1; mode=block
                                                                 X-Frame-Options=SAMEORIGIN
                                                                 X-Android-Received-Millis=1511120052765
                                                                 Access-Control-Allow-Origin=*
                                                             BODY
                                                                 '{"session":{"application_id":58733,"created_at":"2017-11-19T19:34:11Z","id":482417235,"nonce":58840027,"token":"2e9cd0e69518e3db2492a8af8ff54fe31b00e56d","ts":1511120051,"updated_at":"2017-11-19T19:34:11Z","user_id":0,"_id":"482417235"}}'
11-19 21:34:12.835 20464-20464/call.biz.vomer.voip D/QBASDK: =========================================================
                                                             === REQUEST ==== 0cea5934-15f3-400c-ab7a-b4e6ef4ff54e ===
                                                             REQUEST
                                                                 POST https://api.quickblox.com/login.json
                                                             HEADERS
                                                                 QuickBlox-REST-API-Version=0.1.1
                                                                 QB-SDK=Android 3.4.1
                                                                 QB-Token=2e9cd0e69518e3db2492a8af8ff54fe31b00e56d
                                                             PARAMETERS
                                                                 login=vomer_call1
                                                                 password=megahacker
                                                             INLINE
                                                                 POST https://api.quickblox.com/login.json?login=vomer_call1&password=megahacker
11-19 21:34:12.836 20464-20464/call.biz.vomer.voip D/QBASDK: QBSessionListener onSessionCreated: WITHOUT USER
11-19 21:34:13.897 20464-20464/call.biz.vomer.voip D/QBASDK: *********************************************************
                                                             *** RESPONSE *** 0cea5934-15f3-400c-ab7a-b4e6ef4ff54e ***
                                                             STATUS : 202 
                                                             HEADERS
                                                                 X-Runtime=0.018921
                                                                 Cache-Control=no-cache
                                                                 QuickBlox-REST-API-Version=0.1.1
                                                                 Set-Cookie=_mkra_ctxt=8e1f0644f6cd51f5d282e0098be3b7ae--202; path=/; max-age=5; HttpOnly; secure
                                                                 QB-Token-ExpirationDate=2017-11-19 21:34:11 +0000
                                                                 Status=202 Accepted
                                                                 X-Request-Id=cb1611a1-8902-4dbb-abc8-9944b619767d
                                                                 X-Android-Response-Source=NETWORK 202
                                                                 Connection=Close
                                                                 X-Android-Sent-Millis=1511120053365
                                                                 Content-Type=application/json; charset=utf-8
                                                                 Date=Sun, 19 Nov 2017 19:34:13 GMT
                                                                 X-Content-Type-Options=nosniff
                                                                 Server=openresty/1.9.15.1
                                                                 Content-Length=371
                                                                 Strict-Transport-Security=max-age=31536000
                                                                 X-XSS-Protection=1; mode=block
                                                                 X-Frame-Options=SAMEORIGIN
                                                                 X-Android-Received-Millis=1511120053896
                                                                 Access-Control-Allow-Origin=*
                                                             BODY
                                                                 '{"user":{"id":36557657,"owner_id":74506,"full_name":null,"email":null,"login":"vomer_call1","phone":null,"website":null,"created_at":"2017-11-04T15:24:50Z","updated_at":"2017-11-19T19:24:06Z","last_request_at":"2017-11-19T19:34:13Z","external_user_id":null,"facebook_id":null,"twitter_id":null,"blob_id":null,"custom_data":null,"twitter_digits_id":null,"user_tags":null}}'
11-19 21:34:13.936 20464-20464/call.biz.vomer.voip D/QBASDK: QBSessionListener onSessionUpdated
11-19 21:34:13.939 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.app.JobIntentService$JobServiceEngineImpl>
11-19 21:34:13.940 20464-20464/call.biz.vomer.voip I/art: Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.app.JobIntentService$JobServiceEngineImpl>
11-19 21:34:13.996 20464-20464/call.biz.vomer.voip D/QBASDK: SubscribeService: SubscribeService created
11-19 21:34:14.008 20464-20791/call.biz.vomer.voip D/QBASDK: SubscribeService: onHandleIntent start: register to pushes
11-19 21:34:14.037 20464-20791/call.biz.vomer.voip D/QBASDK: SubscribeService: Your meta-data are not set, auto push subscribe unable
11-19 21:34:14.053 20464-20464/call.biz.vomer.voip D/QBASDK: SubscribeService: Service onDestroy
11-19 21:34:14.322 20464-20464/call.biz.vomer.voip D/stas:  authorization = a0:32:99:9a:7c:30
11-19 21:34:14.432 20464-20464/call.biz.vomer.voip D/stas:  {"device":1,"macAddress":"a0:32:99:9a:7c:30","number":"380939474939","token":"fP2PpAMwMXo:APA91bG3neuszzVZbKZdAINlcHaRV0fUs0O6eqvKGM3HhZleC_iVAEvIGddLc3vf5sBae2ZfZ7mQj7VGrZI1KfQ4sakf56_xoj1lSGbwhpDBFJeoO86ILSYa2gLzlWGxhsa5jQ9pRMe4","userID":1}
11-19 21:34:14.704 20464-20464/call.biz.vomer.voip I/PermissionHelper: PERMISSION: Permission Granted
11-19 21:34:15.251 20464-20487/call.biz.vomer.voip I/art: Background partial concurrent mark sweep GC freed 139793(5MB) AllocSpace objects, 7(160KB) LOS objects, 39% free, 15MB/25MB, paused 4.038ms total 190.704ms
11-19 21:34:15.384 20464-20464/call.biz.vomer.voip D/stas:  AllContactLoaded
11-19 21:34:18.823 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.widget.LinearLayout{180ab5cd V.E..... ........ 164,38-656,127 #7f090086 app:id/llInfoBlock}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=232.44937, y[0]=45.74707, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167418118, downTime=167418118, deviceId=7, source=0x1002 }
11-19 21:34:18.823 20464-20464/call.biz.vomer.voip I/View: Touch down dispatch to android.widget.RelativeLayout{29820e82 V.E...C. ........ 0,0-720,165 #7f0900ae app:id/rlContainer}, event = MotionEvent { action=ACTION_DOWN, id[0]=0, x[0]=396.44937, y[0]=83.74707, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167418118, downTime=167418118, deviceId=7, source=0x1002 }
11-19 21:34:18.884 20464-20464/call.biz.vomer.voip I/View: Touch up dispatch to android.widget.RelativeLayout{29820e82 V.E...C. ...p.... 0,0-720,165 #7f0900ae app:id/rlContainer}, event = MotionEvent { action=ACTION_UP, id[0]=0, x[0]=396.44937, y[0]=83.74707, toolType[0]=TOOL_TYPE_FINGER, buttonState=0, metaState=0, flags=0x0, edgeFlags=0x0, pointerCount=1, historySize=0, eventTime=167418185, downTime=167418118, deviceId=7, source=0x1002 }
11-19 21:34:18.918 20464-20464/call.biz.vomer.voip I/PermissionHelper: PERMISSION: Permission Granted
11-19 21:34:18.929 20464-20464/call.biz.vomer.voip D/BaseClient.: Create QBRTCClient INSTANCE
11-19 21:34:18.929 20464-20464/call.biz.vomer.voip D/BaseClient.: init Task Executor
11-19 21:34:18.932 20464-20464/call.biz.vomer.voip D/BaseClient.LooperExecutor: Create looper executor on thread: 1 for QBRTCClient
11-19 21:34:18.932 20464-20464/call.biz.vomer.voip D/BaseClient.LooperExecutor: Request Looper start. On QBRTCClient
11-19 21:34:18.934 20464-20990/call.biz.vomer.voip D/BaseClient.LooperExecutor: Looper thread started.
11-19 21:34:18.935 20464-20990/call.biz.vomer.voip D/BaseClient.LooperExecutor: Looper thread started on thread.11551
11-19 21:34:18.950 20464-20464/call.biz.vomer.voip D/BaseClient.: Call createNewSessionWithOpponents[36557657]conference type=QBConferenceType{value='2'}
11-19 21:34:19.302 20464-20487/call.biz.vomer.voip W/art: Suspending all threads took: 10.519ms
11-19 21:34:19.316 20464-20464/call.biz.vomer.voip D/BaseClient.BaseSession: Create new BaseSession
11-19 21:34:19.328 20464-20464/call.biz.vomer.voip D/BaseClient.BaseSession: onStateChanged to:QB_RTC_SESSION_NEW
11-19 21:34:19.333 20464-20464/call.biz.vomer.voip D/BaseClient.QBRTCSession: isInitiator=true
11-19 21:34:19.335 20464-20464/call.biz.vomer.voip D/AndroidRuntime: Shutting down VM
11-19 21:34:19.341 20464-20487/call.biz.vomer.voip I/art: Background sticky concurrent mark sweep GC freed 107342(4MB) AllocSpace objects, 27(1720KB) LOS objects, 24% free, 19MB/25MB, paused 18.642ms total 116.179ms
11-19 21:34:19.998 20464-20464/call.biz.vomer.voip E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     Process: call.biz.vomer.voip, PID: 20464
                                                                     java.lang.NullPointerException: Attempt to invoke virtual method 'com.quickblox.chat.QBWebRTCSignaling com.quickblox.chat.QBVideoChatWebRTCSignalingManager.getSignaling(int)' on a null object reference
                                                                         at com.quickblox.videochat.webrtc.QBRTCSession.initSignallingWithOpponents(QBRTCSession.java:666)
                                                                         at com.quickblox.videochat.webrtc.QBRTCSession.<init>(QBRTCSession.java:81)
                                                                         at com.quickblox.videochat.webrtc.QBRTCClient.createNewSessionWithOpponents(QBRTCClient.java:127)
                                                                         at call.biz.vomer.voip.base.MainActivity$4.onPermissionGranted(MainActivity.java:313)
                                                                         at com.master.permissionhelper.PermissionHelper.request(PermissionHelper.java:80)
                                                                         at call.biz.vomer.voip.base.MainActivity.startCall(MainActivity.java:304)
                                                                         at call.biz.vomer.voip.screens.main.MainFragment.onContactClicked(MainFragment.java:163)
                                                                         at call.biz.vomer.voip.screens.main.LastContactsAdapter.lambda$onBindViewHolder$0(LastContactsAdapter.java:47)
                                                                         at call.biz.vomer.voip.screens.main.LastContactsAdapter$$Lambda$1.onClick(Unknown Source)
                                                                         at android.view.View.performClick(View.java:4770)
                                                                         at android.view.View$PerformClick.run(View.java:19801)
                                                                         at android.os.Handler.handleCallback(Handler.java:739)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                         at android.os.Looper.loop(Looper.java:135)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5261)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:372)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:939)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:734)
11-19 21:34:21.766 20464-20569/call.biz.vomer.voip I/CrashlyticsCore: Crashlytics report upload complete: 5A11DC9A01AF-0001-4FF0-B2CCE6745929

**Comment by QB-nikolay-zolotarov** [comment]
Hello QuickBlox customer,

This is Nikolay from QuickBlox support.

The issue was closed as it is outdated.

Please check the relevant section of our documentation here:
[https://docs.quickblox.com/docs/android-video-calling](https://docs.quickblox.com/docs/android-video-calling)

Also, please update the SDK to the latest version:
[https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11](https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11)

Additionally, please check our new samples:
[https://docs.quickblox.com/docs/code-samples#video-calling-samples](https://docs.quickblox.com/docs/code-samples#video-calling-samples)

If it is still relevant after reviewing the updated information, feel free to open a new issue.

Have a nice day.
